src/ray/core_worker/store_provider/plasma_store_provider.h

```diff
@@ -135,6 +141,15 @@ class CoreWorkerPlasmaStoreProvider {
   std::mutex store_client_mutex_;
   std::function<Status()> check_signals_;
   std::function<void()> on_store_full_;
+  std::function<std::string()> current_call_site_;
+
+  // Guards the active buffers map. This mutex may be acquired during PlasmaBuffer
+  // destruction.
+  absl::Mutex active_buffers_mutex_;
+  // Mapping of live object buffers to their creation call site. Destroyed buffers are
+  // automatically removed from this list via destructor callback.
+  absl::flat_hash_map<std::pair<ObjectID, PlasmaBuffer *>, std::string> active_buffers_
```

Comment:
Can you add a comment to explain the key? With a note that we should never store a shared_ptr here.